### ISSUE TITLE
Custom table name for Sugar 

### ISSUE THREAD

**Issue by tgalery** [satyan/sugar#298]
Hi I'm creating a custom record called `MessageEntity` that extends the `SugarRecord` object. If I understand this correctly, queries related to this entity will select a table called `message_entity`. Would it be possible to override the table name with a custom one ? If so, how ?

**Comment by whoshuu** [comment]
You have to use annotations if you want custom names for your tables. A quick example would be:

```java
@Table(name="table_name")
public class MessageEntity {
    private long Id;
}
```

Note that I'm no longer extending `SugarRecord`. Sugar will pick up the annotation and create the proper table in the database. Let me know if you have any issues with this.

**Comment by bkawakami** [comment]
This annotation only work for Sugar 1.4

Someone know when 1.4 will be released for gradle and maven?

**Comment by whoshuu** [comment]
There's still quite a lot of work left to do for 1.4.0. My guess is sometime close to the end of this month or mid next month. As for when they'll be released on gradle or maven, I'm not sure. @satyan has been busy, so what I may have to do is host it myself using a different group id. When we're closer to actually releasing 1.4.0, I'll take stock of how many people would be okay with that and then decide from there.

In the meantime, you can build yourself a jar from source by checking out master and running:

    ./gradlew libraryJar

You don't even need `gradle` or anything like that installed. Running this command will take care of that for you.

**Comment by HDANILO** [comment]
@whoshuu, this process will compile 1.4.0? Is 1.4.0 into master already?

**Comment by whoshuu** [comment]
Technically the version of the master branch is 1.3.1. We got lazy with the versioning, but what you see in master will be released in 1.4.0.
